server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -232,8 +233,8 @@ public void cleanupExpiredPools() {
                     subAdapter.deleteSubscription(sub);
                 }
             }
-            deletePool(p);
         }
+        deletePools(toDelete);
```

Comment:
This is bonkers.  Just moving all the deletes outside the loop drops it from an hour to 15 minutes? :clap: 💰